src/components/SRNEntry.vue

```diff
@@ -78,49 +89,73 @@ export default {
       //if multiple SRN's are allowed, then this helps with the activation of the + button, to add more SRN's
       return (
         !this.isSingleEntryOnly &&
-        this.userIDList[0]["userID"].length > this.maxLengthOfSRN - 1
+        this.userIDList.slice(-1)[0]["userID"].length > this.maxLengthOfSRN - 1 &&
+        this.userIDList.length < 10
       );
     },
   },
   methods: {
     isValidSRNFormat(e) {
       //checking to see if each char typed by user is only a number
-      let char = String.fromCharCode(e.keyCode);
-      if (/^[0-9]+$/.test(char)) return true;
+      if (e.keyCode >= 48 && e.keyCode <= 57) return true;
       else e.preventDefault();
     },
-    addField(value, list) {
-      list.push({ value: "" });
+    async addField(list) {
+      //for adding another field, the previously entered ID is validated against the database
+      const userID = parseInt(this.userIDList.slice(-1)[0]["userID"]);
+      if (!isNaN(userID)) {
+        await this.authenticateSRN(userID);
+        //only if the SRN is valid or if the user is entering a SRN for the second time,the loop is entered
+        if (this.isUserValid || this.validateCount > 1) {
+          //setting the flag of the SRN
+          this.userIDList.slice(-1)[0]["valid"] = this.isUserValid;
+          list.push({ userID: "", valid: false });
+          //resetting flags to default for processing next SRN
+          this.isUserValid = false;
+          this.validateCount = 0;
+        }
+      }
     },
     removeField(index, list) {
+      //resetting all messages to default before deleting the input field
+      this.invalidInputMessage = "";
+      this.invalidLoginMessage = "";
+      //edge case: user enters an invalid SRN. Error message is displayed. The user is given another chance. User adds another input field but decided to remove it.
+      // At this point, the variables are reset. So the previously entered SRN is checked again, which should not happen. so setting this validateCount = 2 will bypass this.
+      //Will not affect any other case. The user can either submit or decide to add another field again.
+      this.validateCount = 2;
       list.splice(index, 1);
     },
-    updateValue(event) {
+    updateValue(event, index) {
+      //checks if the 10 characters are entered
       if (event.target.value.length < this.maxLengthOfSRN) {
         this.invalidInputMessage = "Please type 10 numbers / कृपया १० संख्या टाइप करें";
         this.invalidLoginMessage = "";
       } else {
         this.invalidInputMessage = "";
       }
+      //if more than 10 characters are entered, slicing the input to only 10
+      // index tells us which input field is being considered
       if (event.target.value.length > this.maxLengthOfSRN) {
         event.target.value = event.target.value.slice(0, this.maxLengthOfSRN);
-        this.userIDList[0]["userID"] = event.target.value;
+        this.userIDList[index]["userID"] = event.target.value;
```

Comment:
```suggestion
        this.userIDList[index]["userID"] = event.target.value.toString();
```
We have to make sure that when we're storing userIDs in this.userIDList, they are all strings by default